### PR TITLE
feat(GODT-2361): Use simple import encrypter for simple messages

### PR DESCRIPTION
**Pull request by jameshoulahan**

### CODE REVIEW

File: server/messages.go [hunk 1]
```diff
@@ -527,12 +527,10 @@ func (s *Server) importAttachment(userID, messageID string, att *rfc822.Section)
...
 
-	if header.Has("Content-Disposition") {
-		dispType, dispParams, err := mime.ParseMediaType(header.Get("Content-Disposition"))
-		if err != nil {
-			return proton.Attachment{}, fmt.Errorf("failed to parse attachment content disposition: %w", err)
-		}
-
+	if !header.Has("Content-Disposition") {
+		disposition = "attachment"
+		filename = "attachment.bin"
```
Comment (LBeernaertProton) [inline server/messages.go:532]: Are you sure that adding this here won't cause any issues?

File: message_encrypt.go [hunk 2]
```diff
@@ -17,18 +21,202 @@ var CharsetReader func(charset string, input io.Reader) (io.Reader, error)
...
+	var textCount int
+
+	if err := s.Walk(func(s *rfc822.Section) error {
+		if contentType, _, err := s.ContentType(); err != nil {
+			return err
+		} else if contentType.SubType() == "encrypted" {
+			return fmt.Errorf("already encrypted")
+		} else if contentType.SubType() == "signed" {
+			return fmt.Errorf("already signed")
+		} else if contentType.Type() != "text" {
```
Comment (cuthix) [inline message_encrypt.go:46]: Do we have test for invites `text/calendar` ?

File: server/messages.go (hunk 1 above)
Comment (LBeernaertProton) [reply server/messages.go:532]: When we ignored the failed attachments in the current process. The web/IMAP clients are still able to view the contents correctly.

Comment (jameshoulahan) [reply server/messages.go:532]: This is just in the test server. It has no impact on how bridge behaves against the real API.

If we fail to parse the attachments for whatever reason, we'll just fall back to the previous behaviour (full-message encryption).

Comment (LBeernaertProton) [reply server/messages.go:532]: Won't that lead to different behavior between fake and real server?

Comment (LBeernaertProton) [reply server/messages.go:532]: But that's the thing, our API servers don't consider this particular case an error (missing qoutes around file name) and handle this case properly.

Comment (jameshoulahan) [reply server/messages.go:532]: Changes pushed -- now if we fail to read either the content type or the content disposition, we fall back to the old full-message encrypter

**Review by cuthix — APPROVED** [review]

**Pull request merged**